### ISSUE TITLE
[java] AvoidUsingShortType erroneously triggered on overrides of 3rd party methods

### ISSUE THREAD

**Issue by demonfiddler** [pmd/pmd#586]
**Rule Set:** Controversial

**Description:** The rule erroneously reports an error against overrides of 3rd party methods. This makes no sense, as the implementer has no option but to implement the 3rd party API as defined.

**Code Sample demonstrating the issue:**

```java
new org.w3c.dom.traversal.NodeFilter() {
    @Override
    public short acceptNode(@Nullable Node node) {
        return 0; 
    }
}
```

**Running PMD through:** *[CLI | Maven | Eclipse]*
